internal/pkg/dl/secret.go

```diff
@@ -0,0 +1,54 @@
+package dl
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/esapi"
+)
+
+type ExtendedClient struct {
+	*elasticsearch.Client
+	Custom *ExtendedAPI
+}
+
+type ExtendedAPI struct {
+	*elasticsearch.Client
+}
+
+// GET /_fleet/secret/secretId
+func (c *ExtendedAPI) Read(secretId string) (*esapi.Response, error) {
+	req, err := http.NewRequest("GET", "/_fleet/secret/"+secretId, nil)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.Perform(req)
+	if err != nil {
+		return nil, err
+	}
+	return &esapi.Response{StatusCode: res.StatusCode, Body: res.Body, Header: res.Header}, nil
+}
+
+type SecretResponse struct {
+	Id    string
+	Value string
+}
+
+func ReadSecret(client *elasticsearch.Client, secretId string) (string, error) {
+	es := ExtendedClient{Client: client, Custom: &ExtendedAPI{client}}
```

Comment:
the client could be created once and reused for multiple `ReadSecret` calls

Reply:
We should probably use the bulker for this and fetch multiple secrets in a single go with an `mget`. I think the bulker will do this automatically.

By the way - do we cache the policy after the secrets have been replaced? That way we don't have to do this replacement for every agent that needs this same policy update.

Reply:
This is a custom API in elastic, I'm not sure if `mget` works here unless we add a bulk get API to ES.
Not sure about the caching, will check.

Reply:
It doesn't seem that we cache the policy. We could cache the secret values or the whole policy.

Reply:
Forgot about the custom API endpoints. We'll have to test this at scale, but I think if we can cache the whole policy instead of re-fetching and re-building it every time, we can probably defer on adding an mget API.

I believe this caching should be safe since we don't modify policy revisions, only add new ones. One thing we have to be careful about is to not cache the version of the policy that has the agent-specific output credentials inserted. That should happen after the caching layer.

Reply:
Isn't it a concern though to cache the secret values? So that by taking a memory dump, secret values could be exposed.